src/fiatconnect/kyc/KycDenied.tsx

```diff
@@ -1,27 +1,139 @@
-import * as React from 'react'
+import React, { useMemo } from 'react'
 import { StyleSheet, Text } from 'react-native'
 import { SafeAreaView } from 'react-native-safe-area-context'
 import { StackScreenProps } from '@react-navigation/stack'
 import { StackParamList } from 'src/navigator/types'
 import { Screens } from 'src/navigator/Screens'
+import { navigate } from 'src/navigator/NavigationService'
+import { useTranslation } from 'react-i18next'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import { FiatExchangeEvents } from 'src/analytics/Events'
+import { KycStatus as FiatConnectKycStatus } from '@fiatconnect/fiatconnect-types'
+import { convertCurrencyToLocalAmount } from 'src/localCurrency/convert'
+import fontStyles from 'src/styles/fonts'
+import Button, { BtnSizes, BtnTypes } from 'src/components/Button'
+
+import { useSelector } from 'react-redux'
+import { localCurrencyExchangeRatesSelector } from 'src/localCurrency/selectors'
+import getNavigationOptions from 'src/fiatconnect/kyc/getNavigationOptions'
 
 type Props = StackScreenProps<StackParamList, Screens.KycDenied>
 
-// TODO implement designs:
-// (retryable): https://www.figma.com/file/bwIY7peeykI8K4wTrXs4iN/Fiat-Connect?node-id=2887%3A41469
-// (final attempt): https://www.figma.com/file/bwIY7peeykI8K4wTrXs4iN/Fiat-Connect?node-id=2885%3A41683
-function KycDenied(_props: Props) {
-  return (
-    <SafeAreaView style={styles.container}>
-      <Text>KYC Denied</Text>
-    </SafeAreaView>
-  )
+function KycDenied({ route, navigation }: Props) {
+  const { quote, flow, retryable } = route.params
+
+  const exchangeRates = useSelector(localCurrencyExchangeRatesSelector)
```

Comment:
This is a kind of unfortunate pattern and one that complicates things significantly. In order to navigate to the `SelectProvder` screen, we need to derive the fiat amount in the user's local currency from the crypto amount in the quote. Actually, for cash _in_ flows, we'll need to do the opposite (crypto amount from fiat). I think the "right" way to do this would be to show a spinner until `exchangeRates` is set and then set the header/render the component for real, but not sure if we want to do that. Frankly, we could also probably get away with using the "other" amount within the quote object (fiat for cash out, crypto for cash in), since the FC API _is_ required to return both `fiatAmount` and `cryptoAmount` for all quotes. This shouldn't cause any issues (I _think_) since the "other" amount is only ever really used by `SelectProviders` when fetching _external_ quotes, where the precision of fiat/crypto amounts aren't as important.

Wondering if folks have any thoughts..

(I believe @XrayBravoGolf also ran into this issue in an open PR as well.)

Reply:
seems like another solution would be to make the fetchProviders cloud function more flexible, and have it accept just one of the two (fiat or crypto amount). that way we don't need to worry about providing both anymore

Reply:
from a quick look at the cloud function, it looks like Simplex allows either fiat or crypto, and the others appear to require the fiat amount. regardless, we should be able to do the same currency conversion inside the cloud function. but it's kinda interesting that as things currently stand, only the fiat amount is required

Reply:
If we're just getting the fiat amount directly from the quote object itself, things are actually pretty simple here; I agree that it would be useful if `SelectProvider` didn't require both values, but it actually doesn't increase the complexity here if we're just pull both amounts from the quote. If it's worth it, we could refactor the cloud function/wallet such that only a single amount is required when navigating to `SelectProvider` as a followup?

Reply:
alright